### PR TITLE
Kokkos: Remove TriBITS Kokkos subpackages (trilinos/Trilinos#11545)

### PR DESCRIPTION
**Pull request by bartlettroscoe**
These are the same commits for Kokkos as part of Trilinos PR:

* https://github.com/trilinos/Trilinos/pull/11808

Therefore, this PR must be merged before Kokkos is snapshotted into Trilinos again after the merge of trilinos/Trilinos#11808 to Trilinos 'develop'.

### CODE REVIEW

**Comment by bartlettroscoe** [comment]
Hum, some of these GHA jobs don't seem to be testing the correct version of the code.  For example, looking at the job:

* https://github.com/kokkos/kokkos/actions/runs/4888328263/jobs/8725881648?pr=6104

it shows the commit being tested as the merge commit 0690c1b8cf0e37d5a20d923ba23a99d53c7abbfc here:

* https://github.com/kokkos/kokkos/actions/runs/4888328263/jobs/8725881648?pr=6104#step:5:488

The job reports the failure:

```
-- Configuring done (6.7s)
CMake Error at core/unit_test/CMakeLists.txt:1129 (file):
  Error evaluating generator expression:

    $<TARGET_FILE_DIR:KokkosCore_UnitTest_DeviceAndThreads>

  No target "KokkosCore_UnitTest_DeviceAndThreads"


-- Generating done (0.3s)
CMake Generate step failed.  Build files cannot be regenerated correctly.
Error: Process completed with exit code 1.
```

at:

* https://github.com/kokkos/kokkos/actions/runs/4888328263/jobs/8725881648?pr=6104#step:11:137

But if you look at that line of code `core/unit_test/CMakeLists.txt:1129` for that commit it shows:

https://github.com/kokkos/kokkos/blob/0690c1b8cf0e37d5a20d923ba23a99d53c7abbfc/core/unit_test/CMakeLists.txt#L1129-L1133

Where the heck is `KokkosCore_UnitTest_DeviceAndThreads` coming from?

Can someone please help to explain this?

**Comment by dalg24** [comment]
We test the merge commit when there is no conflict but this is a red herring.  I pulled your branch and it yields that error which I also don't understand at this time.

**Comment by dalg24** [comment]
I fixed the issue.

This PR renames our targets and executable names
```
KokkosCore_UnitTest_Cuda                -> Kokkos_CoreUnitTest_Cuda
KokkosAlgorithms_UnitTest_RandomAndSort -> Kokkos_AlgorithmsUnitTest_RandomAndSort
etc.
```
Note that in the current version `KokkosSimd_UnitTest_SIMD -> Kokkos_UnitTest_SIMD` (Simd package name is dropped).

One thing that we need to discuss is that this currently implies a departure from the exe names produced by the makefiles.

**Review by crtrott — APPROVED** [review]
I am ok with the name change. I also don't see a pressing reason to change the Makefile System test names. All in all this gets rid of a bunch of cmake so that sounds good to me.

**Review by masterleinad — COMMENTED** [review]
Apart from the question of renaming tests, this looks good to me.

File: core/unit_test/TestDeviceAndThreads.py [hunk 1]
```diff
@@ -18,8 +18,8 @@
 import unittest
 import subprocess
 
-PREFIX = "$<TARGET_FILE_DIR:KokkosCore_UnitTest_DeviceAndThreads>"
-EXECUTABLE = "$<TARGET_FILE_NAME:KokkosCore_UnitTest_DeviceAndThreads>"
+PREFIX = "$<TARGET_FILE_DIR:Kokkos_CoreUnitTest_DeviceAndThreads>"
+EXECUTABLE = "$<TARGET_FILE_NAME:Kokkos_CoreUnitTest_DeviceAndThreads>"
```
[the remark this answers is not in the document]
Comment (bartlettroscoe) [reply core/unit_test/TestDeviceAndThreads.py:22]: FYI: I went ahead and applied this commit on the Trilinos side in:

* https://github.com/trilinos/Trilinos/pull/11808/commits/df12adb02da7f2d8a1b90a5acb5a1dcd3bff9b96

in PR:

* https://github.com/trilinos/Trilinos/pull/11808

**Review by dalg24 — APPROVED** [review]
Fine with it but want to wait for CI

**Comment by dalg24** [comment]
ctest reports 100% of the tests passed in the SYCL build but it somehow reports failure.  I will ignore.

**Comment by ndellingwood** [comment]
I added an entry to the CHANGELOG 4.1.0 on #5902 listed under 

```
Build System Changes
Incompatibilities (i.e. breaking changes)
```

**Pull request merged**